packages/sit-onyx/src/components/OnyxSlider/OnyxSlider.vue

```diff
@@ -0,0 +1,725 @@
+<script setup lang="ts">
+import { createSlider } from "@sit-onyx/headless";
+import { iconMinusSmall, iconPlusSmall } from "@sit-onyx/icons";
+import { computed, toRefs } from "vue";
+import { useDensity } from "../../composables/density.js";
+import { getFormMessages, useCustomValidity } from "../../composables/useCustomValidity.js";
+import { useErrorClass } from "../../composables/useErrorClass.js";
+import {
+  SKELETON_INJECTED_SYMBOL,
+  useSkeletonContext,
+} from "../../composables/useSkeletonState.js";
+import { useVModel } from "../../composables/useVModel.js";
+import { useRootAttrs } from "../../utils/attrs.js";
+import { FORM_INJECTED_SYMBOL, useFormContext } from "../OnyxForm/OnyxForm.core.js";
+import OnyxFormElement from "../OnyxFormElement/OnyxFormElement.vue";
+import OnyxIconButton from "../OnyxIconButton/OnyxIconButton.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import OnyxTooltip from "../OnyxTooltip/OnyxTooltip.vue";
+import type { OnyxSliderProps } from "./types.js";
+
+const clamp = (value: number, min: number, max: number) => Math.min(Math.max(value, min), max);
+
+/**
+ * Adjusting the position for marks with proper edge offset to prevent overflow because of rounding.
+ * For marks at 0% and 100%, applies a 0.25rem offset to keep them within bounds.
+ */
+const adjustMarkPosition = (percentage: number): string => {
+  if (percentage <= 0) {
+    return "calc(0.25rem)";
+  }
+  if (percentage >= 100) {
+    return "calc(100% - 0.25rem)";
+  }
+  return `${percentage}%`;
+};
+
+const props = withDefaults(defineProps<OnyxSliderProps>(), {
+  min: 0,
+  max: 100,
+  step: 1,
+  shiftStep: 10,
+  marks: false,
+  modelValue: (props) => [props.min ?? 0] as number[],
```

Comment:
```suggestion
  modelValue: (props) => {
    const { min = 0, max = 100 } = props;
    return [(max - min) / 2];
  },
```

The default value should be in the middle of min/max :)
@JoCa96 Since sliders logically always require a value, should we instead of defining a default value ourselves just make the modelValue property required? Otherwise we might set a default value internally that the user is not aware of which might cause errors in some edge cases 

Also the min/max prop could change dynamically so we wouldn't need to make sure that our default value updates when the min/max properties are changed